client-mr/core/src/main/java/org/apache/hadoop/mapred/SortWriteBufferManager.java

```diff
@@ -64,7 +64,7 @@ public class SortWriteBufferManager<K, V> {
   private final Counters.Counter mapOutputRecordCounter;
   private long uncompressedDataLen = 0;
   private long compressTime = 0;
-  private final long taskAttemptId;
+  private final int taskAttemptId;
```

Comment:
I am not sure about restricting `taskAttemptIds` to `int` in such places.

Here is the situation:
1. Spark, Tez and MR provide us with `long` task attempt ids (for Tez and MR, (taskId, attemptId) constitutes a `long` task attempt id, which we restrict to `int` for similar reasons as in 2.)
2. for the purpose of the block id, we limit those `long` task attempt ids to `int`, since we allow only less that 32 bits for it
3. the task attempt id retrieved from the block id is `int` because of that
4. still, all other places could continue to work with `long` task attempt ids if that makes no difference for that code, up-casting `int` task attempt ids to `long` does not harm, as long as the code works with `long`.

This allows to support truly `long` task attempt ids without reverting such code changes in the future.

@zuston @jerqi @zhengchenyu what do you think?

Reply:
Preferring using long for some common methods.